src/components/buttons/button.tsx

```diff
@@ -1,29 +1,45 @@
 import React from 'react';
 import styled from 'styled-components';
-import { TextStyles } from '../typography/typography-v2';
+import { Icon, IconSize } from '../icon/icon';
+import { resetButton } from '../utils';
+import { lighten } from 'polished';
 
 const StyledButton = styled.button`
-  height: 40px;
+  ${resetButton};
+  color: #ffffff;
+  font-size: 16px;
+  font-weight: bold;
+  line-height: 40px;
+  border-radius: 30px;
+  vertical-align: middle;
+  cursor: pointer;
 
   background: linear-gradient(275.41deg, #543fd7 0%, #2756fd 100%);
-  border-radius: 30px;
-  border: none;
 
   &:hover {
     background: #3e61ee;
   }
+
+  &:active {
+    background: ${lighten(0.1, '#3e61ee')};
+  }
 `;
 
-const StyledIcon = styled.span`
-  padding-left: 24px;
-  padding-right: 16px;
+const StyledIcon = styled(Icon)`
+  padding-right: 10px;
+  // pull the fixed size icon closer to the text
+  margin-left: -4px;
+  /**
+    This is a magic number to match the vertical middle of the button
+    with the icon which has a fixed bounding box which won't allow us to align
+    it based on the actual shape of the icon.
+   */
+  vertical-align: -18%;
 `;
 
 const ButtonText = styled.span`
   padding-left: 16px;
-
-  ${TextStyles.toplineR}
```

Comment:
I chose to remove this as it includes a line height and I moved it to the root of the component so that the icon can inherit the size and color aspects.

Reply:
What's the point of having these helpers then? Why not use it and overwrite what you need to overwrite, like the `line-height` instead of copying all the styles in this helper?

Reply:
I thought the button component is a very special component but you might be right, I got carried away I guess.